needle.pm

```diff
@@ -281,23 +260,20 @@ sub get_property_value {
     return undef;
 }
 
-sub TO_JSON {
-    my ($self) = @_;
-
+sub TO_JSON ($self) {
     my %hash = map { $_ => $self->{$_} } qw(tags properties area file png unregistered name);
     return \%hash;
 }
 
-sub wanted_ {
+sub wanted_ () {
     return unless (m/.json$/);
     my $needle = needle->new($File::Find::name);
     $needles{$needle->{name}} = $needle if $needle;
 }
 
-sub default_needles_dir { "$bmwqemu::vars{PRODUCTDIR}/needles" }
+sub default_needles_dir () { "$bmwqemu::vars{PRODUCTDIR}/needles" }
 
-sub init {
-    my $init_needles_dir = ($bmwqemu::vars{NEEDLES_DIR} // default_needles_dir);
+sub init ($init_needles_dir = $bmwqemu::vars{NEEDLES_DIR} // default_needles_dir) {
```

Comment:
So it's now possible to pass `init` an argument? Was that intended?

Reply:
hm, honestly by now I don't remember anymore. But I think it's ok if the variable can be initialized from outside